contrib/babelfishpg_common/src/collation.c

```diff
@@ -1084,9 +1112,15 @@ get_server_collation_oid_internal(bool missingOk)
 
 Oid BABELFISH_CLUSTER_COLLATION_OID()
 {
-
 	if (sql_dialect == SQL_DIALECT_TSQL)
 	{
+		/* 
+		 * If upgrade is going on then we should use oid corresponding to 
+		 * babelfishpg_tsql.restored_server_collation_name.
+		 */
+		if (babelfish_restored_server_collation_name != NULL)
+			return get_collation_oid_internal(babelfish_restored_server_collation_name);
```

Comment:
Previously, `BABELFISH_CLUSTER_COLLATION_OID` always returned valid oid. Is there any possibility of returning invalid oid here?

Reply:
It should not return invalid OID ideally. There are reasons for it, for example, babelfish_restored_server_collation_name would always be valid babelfish collation name or NULL because we uses the same validity function as server_collation_name to validate this GUC also.